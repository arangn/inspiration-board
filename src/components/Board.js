import React, { Component } from 'react';
import PropTypes from 'prop-types';
import axios from 'axios';

import './Board.css';
import Card from './Card';
import NewCardForm from './NewCardForm';
import CARD_DATA from '../data/card-data.json';

class Board extends Component {
  constructor() {
    super();

    this.state = {
      cards: [],
    };
  }

  addCard = (text, emoji) => {
    axios.post(`https://inspiration-board.herokuapp.com/boards/Naheed/cards?text=${text}&emoji=${emoji}`)
      .then((response) => {
        axios.get('https://inspiration-board.herokuapp.com/boards/Naheed/cards')
        .then((response) => {
          this.setState({ cards: response.data });
        })

        // let updatedData = this.state.cards;
        // updatedData.push();
        // this.setState({cards: updatedData});
      })
      .catch((error) => {
        this.setState({ error: error.message });
      });
  }

  deleteCard = (id) => {
    axios.delete(`https://inspiration-board.herokuapp.com/cards/${id}`, id)
    .then((response) => {
      axios.get('https://inspiration-board.herokuapp.com/boards/Naheed/cards')
      .then((response) => {
        this.setState({ cards: response.data });
      })
      // let updatedData = this.state.cards;
      // updatedData.filter((card) => {
      //   return id != card.id;
      // });
      // this.setState({cards: updatedData});
    })
    .catch((error) => {
      this.setState({ error: error.message });
    });
  }

  componentDidMount() {
    axios.get('https://inspiration-board.herokuapp.com/boards/Naheed/cards')
    .then((response) => {
      this.setState({ cards: response.data });
    })
    .catch((error) => {
      this.setState({ error: error.message });
    });
  }

  render() {
    const allCards = this.state.cards.map((card) => {
      return <Card
        key={card.card.id}
        id={card.card.id}
        text={card.card.text}
        emoji={card.card.emoji}
        deleteCardCallback={this.deleteCard}
        addCardCallback={this.addCard}/>
    });

    return (
      <div>
        <div>
          <h3 className="new-card-form__header">Make a new card</h3>
          <NewCardForm cards={this.state.cards} addCardCallback={this.addCard}/>
        </div>
        <ul>
          {allCards}
        </ul>
      </div>
    )
  }
}

Board.propTypes = {

};

export default Board;
