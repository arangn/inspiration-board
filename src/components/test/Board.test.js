import React from 'react';
import Board from '../Board';
import { shallow, mount } from 'enzyme';

const CARDS = [
  {
    name: "test card 1",
    emoji: "emoji"
  },
  {
    name: "test card 2",
    emoji: "emoji"
  }
];

describe('Board', () => {
  test('that it matches an existing snapshot', () => {
    // First Mount the Component in the testing DOM
    // Arrange

    const cards = CARDS;
    const wrapper = shallow(
      <Board
        cards={cards}
        />
    );

    // Assert that it looks like the last snapshot
    expect(wrapper).toMatchSnapshot();
  });

  test('renders without crashing even with no cards', () => {
    // First Mount the Component in the testing DOM
    // Arrange

    const cards = [];
    const wrapper = shallow(
      <Board
        cards={cards}
        />
    );

    // Assert that it looks like the last snapshot
    expect(wrapper).toMatchSnapshot();
  });

  test('successfully deep mounts', () => {
    // First Mount the Component in the testing DOM
    // Arrange
    const wrapper = mount(
      <Board
        cards={CARDS}
        />
    );

    // Assert that it looks like the last snapshot
    expect(wrapper).toMatchSnapshot();

    wrapper.unmount();
  });
});
