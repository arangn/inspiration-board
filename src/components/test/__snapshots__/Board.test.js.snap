// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board renders without crashing even with no cards 1`] = `
<div>
  <div>
    <NewCardForm
      addCardCallback={[Function]}
      cards={Array []}
    />
  </div>
  <ul />
</div>
`;

exports[`Board successfully deep mounts 1`] = `
<Board
  cards={
    Array [
      Object {
        "emoji": "emoji",
        "name": "test card 1",
      },
      Object {
        "emoji": "emoji",
        "name": "test card 2",
      },
    ]
  }
>
  <div>
    <div>
      <NewCardForm
        addCardCallback={[Function]}
        cards={Array []}
      >
        <div
          className="new-card-form"
        >
          <h3
            className="new-card-form__header"
          >
            Make a new card
          </h3>
          <form
            className="new-card-form__form"
            onSubmit={[Function]}
          >
            <div>
              <label
                className="new-card-form__form-label"
                htmlFor="Text"
              />
              <input
                className="new-card-form__form-textarea"
                name="text"
                onChange={[Function]}
                placeholder="Inspiration"
                value=""
              />
            </div>
            <div>
              <label
                className="new-card-form__form-label"
                htmlFor="Emoji"
              />
              <select
                className="new-card-form__form-select"
                name="emoji"
                onChange={[Function]}
              >
                <option
                  disabled={true}
                  hidden={true}
                  selected={true}
                  value=""
                >
                  Emoji
                </option>
                <option
                  key="0"
                  value=""
                />
                <option
                  key="1"
                  value="heart_eyes"
                >
                  😍
                </option>
                <option
                  key="2"
                  value="beer"
                >
                  🍺
                </option>
                <option
                  key="3"
                  value="clap"
                >
                  👏
                </option>
                <option
                  key="4"
                  value="sparkling_heart"
                >
                  💖
                </option>
                <option
                  key="5"
                  value="heart_eyes_cat"
                >
                  😻
                </option>
                <option
                  key="6"
                  value="dog"
                >
                  🐶
                </option>
              </select>
            </div>
            <div>
              <input
                className="new-card-form__form-button"
                type="submit"
                value="Submit Card"
              />
            </div>
          </form>
        </div>
      </NewCardForm>
    </div>
    <ul />
  </div>
</Board>
`;

exports[`Board that it matches an existing snapshot 1`] = `
<div>
  <div>
    <NewCardForm
      addCardCallback={[Function]}
      cards={Array []}
    />
  </div>
  <ul />
</div>
`;
